File: jna/src/main/java/dark/bbox_t_container.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;
/*
struct bbox_t_container {
        bbox_t candidates[C_SHARP_MAX_OBJECTS];
        };
 */
public class bbox_t_container extends Structure implements Structure.ByReference {
    public bbox_t[] candidates = new bbox_t[1000];

    public bbox_t_container() {
        super();
    }

    public bbox_t_container(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("candidates");
    }
}

File: jna/src/main/java/dark_kotlin/bbox_t_container.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure

class bbox_t_container : Structure, Structure.ByReference {
    val candidates = arrayOfNulls<bbox_t>(1000)

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("candidates")
    }
}
